Utilities/Thread.h

```diff
@@ -631,6 +631,24 @@ class named_thread final : public Context, result_storage<Context>, thread_base
 		return *this;
 	}
 
+	// Assign aborting state forcefully and join thread
+	void abort_and_join(bool abort, bool join)
+	{
+		if (abort)
+		{
+			operator=(thread_state::aborting);
+
+			if (!join)
```

Comment:
It would make more sense if the join call below was locked behind that flag instead of only checking it in here